db/external_sst_file_ingestion_job.cc

```diff
@@ -538,9 +538,39 @@ Status ExternalSstFileIngestionJob::AssignLevelAndSeqnoForIngestedFile(
     // if our file can fit in it
     if (IngestedFileFitInLevel(file_to_ingest, lvl)) {
       target_level = lvl;
+    } else if (lvl == cfd_->NumberLevels() - 1) {
+      ROCKS_LOG_INFO(db_options_.info_log,
+                     "can't ingest into bottom level due to range overlap %d",
+                     cfd_->NumberLevels() - 1);
+    }
+  }
+
+  auto origin_target_level = target_level;
+  while (target_level >= vstorage->base_level()) {
+    if (vstorage->MaxStaticBytesForLevel(target_level) <=
+            file_to_ingest->file_size + vstorage->NumLevelBytes(target_level) ||
+        !IngestedFileFitInLevel(file_to_ingest, target_level)) {
+      ROCKS_LOG_INFO(
+          db_options_.info_log, "ingest bytes %" PRIu64 " + %" PRIu64
+                                " exceed level size %" PRIu64 " for level %d",
+          file_to_ingest->file_size, vstorage->NumLevelBytes(target_level),
+          vstorage->MaxBytesForLevel(target_level), target_level);
+      target_level--;
+    } else {
+      break;
     }
   }
- TEST_SYNC_POINT_CALLBACK(
+
+  if (target_level == 0) {
+    target_level = origin_target_level;
+    ROCKS_LOG_INFO(db_options_.info_log, "ingest to origin level %d",
+                   origin_target_level);
+  } else if (target_level < vstorage->base_level()) {
+    target_level = 0;
+    ROCKS_LOG_INFO(db_options_.info_log, "ingest before base level %d",
+                   vstorage->base_level());
+  }
```

Comment:
why inconsistent?

Reply:
Can you explain the first branch's intention? Is it here to avoid unnecessary ingest to L0 causing further WA in intra-L0 compaction? If that's the case then the second branch contradicts such intention.

Reply:
you're right. And it a remnant of the previous version of the design. I will remove it too.